### PR TITLE
token bearer configurable

### PR DESCRIPTION
**Pull request by priyanka-TL**

### CODE REVIEW

File: src/envVariables.js [hunk 1]
```diff
@@ -32,6 +32,11 @@ let enviromentVariables = {
 		message: 'Required api doc url',
 		optional: false,
 	},
+	IS_AUTH_TOKEN_BEARER: {
```
Comment (VISHNUDAS-tunerlabs) [inline src/envVariables.js:35]: @priyanka-TL can you please verify if  this variable is not passed from .env. the default value is getting passed correctly

File: src/generics/middleware/authenticator.js [hunk 2]
```diff
@@ -37,7 +38,21 @@ module.exports = async function (req, res, next, token = '') {
...
+	let authHeader = req.headers['x-auth-token']
+	if (isBearerRequired) {
+		const [authType, extractedToken] = authHeader.split(' ')
+		if (authType.toLowerCase() !== 'bearer') {
+			rspObj.errCode = CONSTANTS.apiResponses.TOKEN_INVALID_CODE
+			rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_INVALID_MESSAGE
+			rspObj.responseCode = HTTP_STATUS_CODE['unauthorized'].status
+			return res.status(HTTP_STATUS_CODE['unauthorized'].status).send(respUtil(rspObj))
+		}
+		token = extractedToken?.trim()
```
Comment (VISHNUDAS-tunerlabs) [inline src/generics/middleware/authenticator.js:51]: @priyanka-TL where are we defining this token variable?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on Dec 12 : 16:01

File: src/generics/middleware/authenticator.js (hunk 2 above)
Comment (priyanka-TL) [reply src/generics/middleware/authenticator.js:51]: Its already defined in function argument, so if we tried to define again it cause issue

File: src/envVariables.js (hunk 1 above)
Comment (priyanka-TL) [reply src/envVariables.js:35]: yes its working correctly

File: src/generics/middleware/authenticator.js (hunk 2 above)
Comment (VISHNUDAS-tunerlabs) [reply src/generics/middleware/authenticator.js:51]: @priyanka-TL okey got it. thanks

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]

**Pull request merged**